src/MessageReader.js

```diff
@@ -52,13 +73,16 @@ class StandardTypeReader {
     const len = this.int32();
     const codePoints = new Uint8Array(this.buffer.buffer, this.buffer.byteOffset + this.offset, len);
     this.offset += len;
-    // if the string is relatively short we can use apply
-    // but very long strings can cause a stack overflow due to too many arguments
-    // in those cases revert to a slower iterative string building approach
-    if (codePoints.length < 1000) {
-      return String.fromCharCode.apply(null, codePoints);
+
+    // Use TextDecoder if it is available and supports the "ascii" encoding.
+    if (this._decoderStatus === "NOT_INITIALIZED") {
```

Comment:
I think I'd slightly prefer a null check here if we can, but it doesn't really matter -- the happy path disagrees with this string at the first character. Feel free to ignore.